app/access/api.py

```diff
@@ -42,3 +45,30 @@ def users(request: HttpRequest) -> dict[str, list[UserSchema]]:
     models = User.objects.all()
     responses = [user_to_response(model) for model in models]
     return {"items": responses}
+
+
+@router.post("users", response={201: UserSchema})
+def create_user(request: HttpRequest, user_in: UserSchema) -> UserSchema:
+    """Create the given user and return it.
+
+    Return HTTP status code
+
+        - 201 (Created) if the User was created as expected
+        - 404 (Not Found) if there is no provider with the given provider ID
+        - 409 (Conflict) if there is already a record with the same username
+        - 422 (Unprocessable Content) if there is any other invalid value
+    """
+    provider = get_object_or_404(Provider, id=user_in.provider_id)
+
+    try:
+        user_out = User.objects.create(
+            username=user_in.username,
+            first_name=user_in.first_name,
+            last_name=user_in.last_name,
+            email=user_in.email,
+            provider=provider
+        )
+    except ValidationError as error:
+        raise validation_error_to_http_error(error) from error
```

Comment:
Is this necessary? If so, might there be a way to have this globally, as suggested here https://django-ninja.dev/guides/errors/#override-default-handler ?

Reply:
Yes, I think it's necessary to do something with the exception because otherwise the user would just see a generic "500 Internal Server Error".

I was not aware of the possibility to define custom exception handlers for the whole NinjaAPI object - thanks! I'm going to try that, then.